piptools/utils.py

```diff
@@ -591,36 +591,47 @@ def _validate_config(
     :raises click.NoSuchOption: if config contains unknown keys.
     :raises click.BadOptionUsage: if config contains invalid values.
     """
-    cli_params = {
-        param.name: param
-        for param in click_context.command.params
-        if param.name is not None
+    from piptools.scripts.compile import cli as compile_cli
+    from piptools.scripts.sync import cli as sync_cli
+
+    compile_cli_params = {
+        param.name: param for param in compile_cli.params if param.name is not None
+    }
+
+    sync_cli_params = {
+        param.name: param for param in sync_cli.params if param.name is not None
     }
 
+    all_keys = set(compile_cli_params) | set(sync_cli_params)
+
     for key, value in config.items():
-        # Validate unknown keys
-        if key not in cli_params:
-            possibilities = difflib.get_close_matches(key, cli_params.keys())
+        # Validate unknown keys in both compile and sync
+        if key not in all_keys:
+            possibilities = difflib.get_close_matches(key, all_keys)
             raise click.NoSuchOption(
                 option_name=key,
                 message=f"No such config key {key!r}.",
                 possibilities=possibilities,
                 ctx=click_context,
             )
 
-        # Validate invalid values
-        param = cli_params[key]
-        try:
-            param.type_cast_value(value=value, ctx=click_context)
-        except Exception as e:
-            raise click.BadOptionUsage(
-                option_name=key,
-                message=(
-                    f"Invalid value for config key {key!r}: {value!r}.{os.linesep}"
-                    f"Details: {e}"
-                ),
-                ctx=click_context,
-            ) from e
+        # Validate values for both compile and sync
+        for cli_params in (compile_cli_params, sync_cli_params):
+            param = cli_params.get(key)
+            if param is None:
+                continue
+
+            try:
+                param.type.convert(value=value, param=param, ctx=click_context)
```

Comment:
I think you still need to update that after the merge?

```suggestion
                param.type_cast_value(value=value, ctx=click_context)
```

Reply:
Yeah, I messed up with resolving conflicts :/

Reply:
Addressed in f0236945719081f9c012e052f785b28e8048d514